core/pl/stream.pm.sdoc

```diff
@@ -59,6 +59,7 @@ sub rw(&) {my @r = ($_); push @r, $_ while  defined rl && &{$_[0]}; push @q, $_
 sub ru(&) {my @r = ($_); push @r, $_ until !defined rl || &{$_[0]}; push @q, $_ if defined $_; @r}
 sub re(&) {my ($f, $i) = ($_[0], &{$_[0]}); rw {&$f eq $i}}
 BEGIN {ceval sprintf 'sub re%s() {re {%s}}', $_, $_ for 'a'..'l'}
+sub rn($) {my $max_val = shift; my @r = ($_); push @r, $_ while defined rl && (@r < $max_val); push @q, $_ if defined $_; @r} 
```

Comment:
`rl()` has a variant that lets you request N lines (`rl(3)` for instance) -- but it fills out with `undef` if you request more than exist.

`rn` has a subtle bug too: `rl` gets called before `@r < $max_val` is checked, so your last line gets thrown away (I think)